modules/calcite/src/test/java/org/apache/ignite/internal/processors/cache/SessionContextSqlFunctionTest.java

```diff
@@ -0,0 +1,232 @@
+
+            for (List<?> row: rows) {
+                String actSesId = row.get(1).toString();
+
+                assertEquals(sesId, actSesId);
+            }
+        }
+    }
+
+    /** */
+    @Test
+    public void testInsertClause() {
+        for (int i = 0; i < 100; i++) {
+            String sesId = i % 2 == 0 ? "1" : "2";
+
+            Ignite ignApp = ign.withApplicationAttributes(F.asMap(SESSION_ID, sesId));
+
+            ignQuery(ignApp, "insert into PUBLIC.MYTABLE(id, sessionId) values (" + i + ", sessionId());");
+        }
+
+        List<List<?>> res = ignQuery(ign, "select * from PUBLIC.MYTABLE where sessionId = 1");
+
+        assertEquals(50, res.size());
+
+        res = ignQuery(ign, "select * from PUBLIC.MYTABLE where sessionId = 2");
+
+        assertEquals(50, res.size());
+    }
+
+    /** */
+    @Test
+    public void testNestedQuery() {
+        for (int i = 0; i < 100; i++) {
+            String sesId = i % 2 == 0 ? "1" : "2";
+
+            ignQuery(ign, "insert into PUBLIC.MYTABLE(id, sessionId) values (?, ?);", i, sesId);
+        }
+
+        String sesId = "1";
+
+        Ignite ignApp = ign.withApplicationAttributes(F.asMap(SESSION_ID, sesId));
+
+        List<List<?>> rows = ignQuery(ignApp, "select * from PUBLIC.MYTABLE where sessionId = (select sessionId());");
+
+        int size = 0;
+
+        for (List<?> row: rows) {
+            String actSesId = row.get(1).toString();
+
+            assertEquals(sesId, actSesId);
+
+            size++;
+        }
+
+        assertEquals(50, size);
+    }
+
+    /** */
+    @Test
+    public void testOverwriteApplicationAttributes() {
+        Ignite ignApp = ign;
+
+        for (int i = 0; i < 100; i++) {
+            String sesId = i % 2 == 0 ? "1" : "2";
+
+            ignApp = ignApp.withApplicationAttributes(F.asMap(SESSION_ID, sesId));
+
+            ignQuery(ignApp, "insert into PUBLIC.MYTABLE(id, sessionId) values (" + i + ", sessionId());");
+        }
+
+        List<List<?>> res = ignQuery(ign, "select * from PUBLIC.MYTABLE where sessionId = 1");
+
+        assertEquals(50, res.size());
+
+        res = ignQuery(ign, "select * from PUBLIC.MYTABLE where sessionId = 2");
+
+        assertEquals(50, res.size());
+    }
+
+    /** */
+    @Test
+    public void testMultithreadApplication() throws Exception {
+        String sesId = "1";
+
+        Ignite ignApp = ign.withApplicationAttributes(F.asMap(SESSION_ID, sesId));
+
+        IgniteInternalFuture<?> insertFut = multithreadedAsync(() -> {
+            for (int i = 0; i < 100; i++)
+                ignQuery(ignApp, "insert into PUBLIC.MYTABLE(id, sessionId) values (" + i + ", sessionId());");
+        }, 1, "insert");
+
+        insertFut.get(getTestTimeout());
+
+        List<List<?>> res = ignQuery(ign, "select * from PUBLIC.MYTABLE where sessionId = " + sesId);
+
+        assertEquals(100, res.size());
+    }
+
+    /** */
+    private List<List<?>> ignQuery(Ignite ign, String sql, Object... args) {
```

Comment:
`ignQuery` - not a common abbreviation for Ignite. Perhaps just `query`? Or just `sql`?